src/mono/mono/metadata/threads.c

```diff
@@ -1247,13 +1243,11 @@ start_wrapper_internal (StartInfo *start_info, gsize *stack_ptr)
 
 	THREAD_DEBUG (g_message ("%s: (%" G_GSIZE_FORMAT ") Start wrapper terminating", __func__, mono_native_thread_id_get ()));
 
-#if defined(HOST_BROWSER) && !defined(DISABLE_THREADS)
-	if (returns_to_js_event_loop) {
-		/* if the thread wants to stay alive, don't clean up after it */
-		if (emscripten_runtime_keepalive_check())
+	if (G_UNLIKELY (external_eventloop)) {
```

Comment:
Is the G_UNLIKELY here important or necessary? Just curious

Reply:
Neither

Reply:
It is also a hint to compilers that supports it to put the branch out of line assisting the branch predictors heuristics, so it should then end up with a conditional jmp to an out of line code block when true, but branch predictor will predict it to not be taken (normal default heuristics for forward branches).